### PR TITLE
DEVOPS-1843 Fix US DEV Web Vault deploys one commit behind

### PR DESCRIPTION
**Pull request by urbinaalex17**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The `build-web` and `deploy-web` workflows work together to deploy the correct latest commit hash from main into US DEV

## Code changes

- **.github/workflows/build-web.yml:** Sends the exact run_id that `.github/workflows/deploy-web.yml` workflow must use to download the artifacts.
- **.github/workflows/deploy-web.yml:**  Accepts the run_id that `.github/workflows/build-web.yml` sent.

Commits: `7603a6a301f84e00b3fd67999ee96978dac508c7`, `b15adf42e990fc86897f1eb4463d911847d0e631` and `c3354278a5e0f69915c128219bf91818cf476897` must be reverted before merging this PR. 

## Screenshots

image

### Tests:

- [Build Web](https://github.com/bitwarden/clients/actions/workflows/build-web.yml) triggered [Deploy Web Vault](https://github.com/bitwarden/clients/actions/workflows/deploy-web.yml)
- [Deploy Web Vault](https://github.com/bitwarden/clients/actions/workflows/deploy-web.yml) is working as expected when the workflow is run manually.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: .github/workflows/deploy-web.yml [hunk 1]
```diff
@@ -27,6 +27,10 @@ on:
         description: "Debug mode"
         type: boolean
         default: true
+      build-web-run-id:
+        description: "Build-web workflow Run ID to use for artifact download"
```
Comment (mimartin12) [inline .github/workflows/deploy-web.yml:31]: 🎨  Descriptions could be updated to be the same. https://github.com/bitwarden/clients/pull/8458/files#diff-875a32a9bd9123a01ecb0b1802809c712a8a04f695e979c262eb9380e6359efcR53

**Review by mimartin12 — CHANGES_REQUESTED** [review]
Just a small item on this, but could you please revert the test code prior to another review?

File: .github/workflows/deploy-web.yml (hunk 1 above)
Comment (urbinaalex17) [reply .github/workflows/deploy-web.yml:31]: @mimartin12, I reverted the test code!

File: .github/workflows/deploy-web.yml [hunk 2]
```diff
@@ -46,6 +50,10 @@ on:
         description: "Debug mode"
         type: boolean
         default: true
+      build-web-run-id:
+        description: "Build-web workflow specific Run ID"
```
Comment (mimartin12) [inline .github/workflows/deploy-web.yml:54]:
```suggestion
        description: "Build-web workflow Run ID to use for artifact download"
```

File: .github/workflows/deploy-web.yml (hunk 1 above)
Comment (mimartin12) [reply .github/workflows/deploy-web.yml:31]: I made a direct suggestion here. https://github.com/bitwarden/clients/pull/8458/files#r1537907933

File: .github/workflows/deploy-web.yml (hunk 2 above)
Comment (urbinaalex17) [reply .github/workflows/deploy-web.yml]: Thanks, @mimartin12 !

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**